docs/library/rp2.rst

```diff
@@ -71,6 +71,79 @@ For running PIO programs, see :class:`rp2.StateMachine`.
     This exception is raised from `asm_pio()` or `asm_pio_encode()` if there is
     an error assembling a PIO program.
 
+Each state machine has a set of instructions- which allow for a wide range of
+behaviourial or location changes to the data. They are written in assembly
+language which is later used in a MicroPython program to perform specific
+tasks. These instructions are -
+
+jmp(cond, lebel=None)
+	condition : 0-7
+	<delay_value> : 0-31
+	target : instruction offset of label within the program
+    see sec 3.4.2 of RPi docs for details
+
+wait(polarity, src, index)
+     polarity: 0-1
+     src: gpio, pin, irq
+     index: 0-31
+     wait for high/low on pin 0-31 or irq 0-31
+     see sec 3.4.3 of RPi docs for details
+
+in(src, data)
+	src : pin, X, Y, ISR, OSR(all 0-31)
+	data : bit count(0-31)
+    see sec 3.4.4 of RPi docs for details
+
+out(dist,data)
+	dist : pin, X, Y, PINDIRS, AC, ISR, OSR(all 0-31)
+	data : bit count(0-31)
+    see sec 3.4.5 of RPi docs for details
+
+push(value = 0, value = 2)
+	block = 0 : no block
+    block = 1 : block
+    see sec 3.4.6 of RPi docs for details
+
+pull(value = 0, value = 2)
+	pull(if empty) : 1 (default = 0)
+	block : 1 (default = 1)  
+	no-block = 0
+    see sec 3.4.7 of RPi docs for details
+
+mov(dist, src)
+	dist : pin, X, Y, ISR, OSR, PC (all 0-31)
+	src : pin, X, Y, ISR, OSR (all 0-31)
+    see sec 3.4.8 of RPi docs for details
+
+set(dest, data)
+	dest : X, Y, PINS, PINDIRS
+	data : value (0-31)
+    see sec 3.4.10 of RPi docs for details
+
+irq(mod, index = None)
+	These are the following forms of this instruction(index is an integer value 
+	from 0-7)
+        * irq(index)
+	* irq(rel(index))
+	* irq(block, index)
+	* irq(block, rel(index))
+	* irq(clear, rel(index))
+	* irq(block | clear, index)
+	* irq(block | clear, rel(index))
```

Comment:
either keep this set of possible ways of writing it, or the below set of 2 forms, but not both